yt/frontends/cf_radial/data_structures.py

```diff
@@ -0,0 +1,269 @@
+        # will be converted to YTArray automatically at a later time.
+        # This includes the cosmological parameters.
+
+        #
+        #   self.unique_identifier      <= unique identifier for the dataset
+        #                                  being read (e.g., UUID or ST_CTIME)
+        self.unique_identifier = int(os.stat(self.parameter_filename)[stat.ST_CTIME])
+        #   self.parameters             <= full of code-specific items of use
+        self.parameters = {}
+
+        coords = self._handle.coords
+        x, y, z = [coords[d] for d in "xyz"]
+
+        self.origin_latitude = self._handle.origin_latitude[0]
+        self.origin_longitude = self._handle.origin_longitude[0]
+
+        #   self.domain_left_edge       <= array of float64
+        dle = [x.min(), y.min(), z.min()]
+        self.domain_left_edge = np.array(dle)
+        #   self.domain_right_edge      <= array of float64
+        dre = [x.max(), y.max(), z.max()]
+        self.domain_right_edge = np.array(dre)
+        #   self.dimensionality         <= int
+        self.dimensionality = 3
+        #   self.domain_dimensions      <= array of int64
+        dims = [self._handle.dims[d] for d in "xyz"]
+        self.domain_dimensions = np.array(dims, dtype="int64")
+        #   self.periodicity            <= three-element tuple of booleans
+        self._periodicity = (False, False, False)
+        #   self.current_time           <= simulation time in code units
+        self.current_time = float(self._handle.time.values)
+        #
+        # We also set up cosmological information.  Set these to zero if
+        # non-cosmological.
+        #
+        #   self.cosmological_simulation    <= int, 0 or 1
+        #   self.current_redshift           <= float
+        #   self.omega_lambda               <= float
+        #   self.omega_matter               <= float
+        #   self.hubble_constant            <= float
+        self.cosmological_simulation = 0.0
+        self.current_redshift = 0.0
+        self.omega_lambda = 0.0
+        self.omega_matter = 0.0
+        self.hubble_constant = 0.0
+
+    @classmethod
+    def _is_valid(self, filename, *args, **kwargs):
+        # This accepts a filename or a set of arguments and returns True or
+        # False depending on if the file is of the type requested.
+        try:
+            import xarray
+        except ImportError:
+            return False
+
+        try:
+            ds = xarray.open_dataset(filename)
+        except OSError:
+            return False
+
+        try:
+            # handle the different upper/lower case cases
+            conventions = ds.attrs.get("Conventions", "")
+            conventions += ds.attrs.get("conventions", "")
+        except AttributeError:
+            # either attrs does not exist or its not a dictionary, so not valid
+            return False
+
+        if "CF/Radial" in conventions:
+            return True
+
+        return False
```

Comment:
I like compact code 😎 
```suggestion
        return "CF/Radial" in conventions
```